src/Features/DynamicCubemaps.cpp

```diff
@@ -91,11 +91,28 @@ void DynamicCubemaps::DrawSettings()
 					DirectX::ScratchImage image;
 					DX::ThrowIfFailed(CaptureTexture(device, context, tempTexture, image));
 
-					std::string filename = std::format("Data\\Textures\\DynamicCubemaps\\{:.2f}{:.2f}{:.2f}R{:.2f}.dds", settings.CubemapColor.x, settings.CubemapColor.y, settings.CubemapColor.z, settings.CubemapColor.w);
+					bool defaultDynamicCubeMapSavePathSafe = true;
+					try {
+						if (std::filesystem::create_directories(defaultDynamicCubeMapSavePath)) {
+							logger::info("Missing DynamicCubeMap Creator directory created: {}", defaultDynamicCubeMapSavePath);
+						}
+					} catch (const std::exception) {
+						logger::error("Failed to create missing DynamicCubeMap Creator directory: {}", defaultDynamicCubeMapSavePath);
+						defaultDynamicCubeMapSavePathSafe = false;
+					}
 
-					std::wstring wfilename = std::wstring(filename.begin(), filename.end());
-					DX::ThrowIfFailed(SaveToDDSFile(image.GetImages(), image.GetImageCount(), image.GetMetadata(), DirectX::DDS_FLAGS::DDS_FLAGS_NONE, wfilename.c_str()));
+					if (defaultDynamicCubeMapSavePathSafe) {
```

Comment:
Shouldn't it be just a check if it exists?

Reply:
my thought was to not not attempt saving the file if there was a failure in creating a new directory, but allow the rest of the method to continue (.Release etc...)

Was not sure how to use Imgui to inform user, which means silently fails in the UI.

or would it be preferable to just wrap the entire thing in a directory check and do nothing but log if the directory does not exist?

Reply:
I'm thinking the simplest way is to look at the use case. A user is trying to just save a cubemap. So we first solve the CTD, like you mentioned, and then if they try to save and it fails, just some sort of messaging. See the message popup we do for failed shaders.

Reply:
See https://github.com/doodlum/skyrim-community-shaders/blob/dev/src/Menu.cpp#L195-L208

Reply:
kept the directory creation for now. 
perhaps will makes this configurable in a later change.

saves will not be attempted if a file object with the same name exists. 
generic message popup  added as Issue #519 

currently message error message display is tightly coupled with ShaderCompiler